CONTRIBUTING.md

```diff
@@ -26,133 +26,169 @@ We specifically do not merge PRs **without prior issues** that:
 
+This command will produce a `tofu` binary in your current directory, which you can test by running `./tofu --version`.
 
----
+> [!TIP]
+> Replace the `GOOS` and `GOARCH` values with your target platform if you wish to cross-compile.
 
-<!-- MarkdownTOC autolink="true" -->
+#### Building in a container
 
-- [Contributing a Code Change](#contributing-a-code-change)
-- [Working on the Code](#working-on-the-code)
-- [Adding or updating dependencies](#adding-or-updating-dependencies)
-- [Acceptance Tests: Testing interactions with external services](#acceptance-tests-testing-interactions-with-external-services)
-- [Generated Code](#generated-code)
+If you have Docker or Podman installed, you can run the entire build process in a container too:
 
-<!-- /MarkdownTOC -->
+```sh
+docker run \
+  --rm \
+  -v "$PWD":/usr/src/opentofu\
+  -w /usr/src/opentofu golang:1.21.3\
+  GOOS=linux GOARCH=amd64 go build -o tofu -v -buildvcs=false ./cmd/tofu
+```
 
-## Contributing a Code Change
+This will create the `tofu` binary in the current working directory, which you can test by running `./tofu --version`.
 
-In order to contribute a code change, you should fork the repository, make your changes, and then submit a pull request. Crucially, all code changes should be preceded by an issue that you've been assigned to. If an issue for the change you'd like to introduce already exists, please communicate in the issue that you'd like to take ownership of it. If an issue doesn't yet exist, please create one expressing your interest in working on it and discuss it first, prior to working on the code. Code changes without a related issue will generally be rejected.
+> [!TIP]
+> Replace the `GOOS` and `GOARCH` values with your target platform if you wish to cross-compile. 
 
-**⚠️ Important:** Please avoid working on features or refactor without [an `accepted` issue](https://github.com/opentofu/opentofu/issues?q=is%3Aopen+is%3Aissue+label%3Aaccepted). OpenTofu is a large and complex project and every change needs careful consideration. We cannot merge non-bug pull requests without first having a discussion about them, no matter how trivial the issue may seem.
+---
 
-In order for a code change to be accepted, you'll also have to accept the Developer Certificate of Origin (DCO). It's very lightweight, and you can find it [here](https://developercertificate.org). Accepting is accomplished by signing off on your commits, you can do this by adding a `Signed-off-by` line to your commit message, like here:
-```
-This is my commit message
+### Running tests
+
+Similar to builds, you can use the `go test` command to run tests. To run the entire test suite, please run the following command in your OpenTofu source directory:
 
-Signed-off-by: Random Developer <random@developer.example.org>
+```sh
+go test ./...
 ```
-Git has a built-in flag to append this line automatically:
+
+Alternatively, you can also run the `go test` command in the package you are currently working on.
```

Comment:
maybe a link to the docs for more ways people can test (https://pkg.go.dev/cmd/go#hdr-Test_packages) ?